webapp/components/ComponentSlider/ComponentSlider.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div class="Sliders">
+    <slot :name="'header'" />
+
+    <ds-heading v-if="sliderData.sliders[sliderIndex].title" size="h3">
+      {{ sliderData.sliders[sliderIndex].title }}
+    </ds-heading>
+
+    <slot :name="sliderData.sliders[sliderIndex].name" />
+
+    <ds-flex>
+      <ds-flex-item :centered="true">
+        <div
+          v-for="(slider, index) in sliderData.sliders"
+          :key="slider.name"
+          :class="['Sliders__slider-selection', index < sliderIndex && '--confirmed']"
+        >
+          <base-button
+            :class="['selection-dot']"
+            style="float: left"
+            :circle="true"
+            size="small"
+            type="submit"
+            filled
+            :loading="false"
+            :disabled="index > sliderIndex"
+            @click="sliderData.sliderSelectorCallback(index)"
+          />
+        </div>
+      </ds-flex-item>
+      <ds-flex-item>
+        <base-button
+          style="float: right"
+          :icon="sliderData.sliders[sliderIndex].button.icon"
+          type="submit"
+          filled
+          :loading="false"
+          :disabled="!sliderData.sliders[sliderIndex].validated"
+          @click="onNextClick"
+        >
+          {{ sliderData.sliders[sliderIndex].button.title }}
+        </base-button>
+      </ds-flex-item>
+    </ds-flex>
+
+    <slot :name="'footer'" />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'ComponentSlider',
+  props: {
+    sliderData: { type: Object, required: true },
+  },
+  computed: {
+    sliderIndex() {
+      return this.sliderData.sliderIndex // to have a shorter notation
+    },
+  },
+  methods: {
+    async onNextClick() {
+      let success = true
+      if (this.sliderData.sliders[this.sliderIndex].button.sliderCallback) {
+        success = await this.sliderData.sliders[this.sliderIndex].button.sliderCallback()
+      }
+      success = success && this.sliderData.sliders[this.sliderIndex].button.callback(success)
+      if (success && this.sliderIndex < this.sliderData.sliders.length - 1) {
+        this.sliderData.sliderSelectorCallback(this.sliderIndex + 1)
+      }
+    },
```

Comment:
This whole function seems a bit omplicated. Why three functions `sliderCallback`, `callback` and `sliderSelectorCallback` ?

Reply:
We invent a more general slider component and therefore I guess it is not always clear if decisions about what will happen on a button click can always only be done in the deepest level.
Therefore I implement a mechanism to have the possibility to decide at top-level as well or in combination. @ulfgebhardt